discovery_engine_core/web-api/openapi.yaml

```diff
@@ -0,0 +1,160 @@
+openapi: 3.0.3
+info:
+  title: Personalization API
+  version: 1.0.0-rc1
+  description: |-
+    The endpoints can be contacted at `t<name>.xaynet.dev/v1` or `t<name>.xaynet.dev/v1.0`.
+    The `v1` path will always point to the latest `v1.x`.
+    
+    # Auth
+    To authenticate with the API two bearer tokens will be provided. One will work for the
+    personalization endpoints and the other one will allow documents to be added to the system.
+    
+tags:
+  - name: personalization
+    description: Interaction of the user and personalized document
+  - name: ingestion
+    description: Add documents to the system
+paths:
+  /users/{user_id}/documents/personalized:
+    get:
+      tags:
+        - personalization
+      summary: Get personalized document for the user
+      description: |-
+        Returns a map from `document_id` to the score of that document for the
+        specified `user_id`. The score is a value between 0 and 1 where a higher value
+        means that the document matches the preferences of the user better.
+      parameters:
+        - name: user_id
+          in: path
+          description: Id of the user
+          required: true
+          schema:
+            $ref: '#/components/schemas/UserId'
+        - name: count
+          in: query
+          description: Maximum number of personalized documents to return
+          type: integer
+          format: int32
+          minimum: 1
+          maximum: 100
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PersonalizedDocuments'
+        '400':
+          description: invalid user id
+        '404':
+          description: user not found or the user does not have enough interactions
+
+  /users/{user_id}/documents/interacted:
+    post:
+      tags:
+        - personalization
+      summary: Add interaction between a user and a document
+      description: |-
+        The interaction is used to provide personalized documents to the user.
+      parameters:
+        - name: user_id
+          in: path
+          description: Id of the user
+          required: true
+          schema:
+            $ref: '#/components/schemas/UserId'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/UserInteraction'
+      responses:
+        '204':
+          description: successful operation
+        '400':
+          description: invalid request. User or document id is invalid
+        '404':
+          description: user not found
```

Comment:
maybe 404 should be if we don't find a document with specified `document_id`? User interests won't necessarily exist when we first call the endpoint

Reply:
the document id is not part of the URI so I think is not correct to return 404 in that case.
Yes actually 404 does not make sense here, if the user does not exists we implicitly create one.

Reply:
Idd, but it's in the post, so then a 500 would make sense if it doesn't exist

Reply:
> the document id is not part of the URI so I think is not correct to return 404 in that case.

through IMHO it should be part of the URI